### DISCUSSION TITLE
How can I add a Krustlet node

### DISCUSSION THREAD

**Discussion by sunfuze** [abiosoft/colima#452] [Q&A]
I want run a krustlet in a lima vm, as follow guides, new node need an ip address, and lima vm need connect to colima vm. How can i config networks make this work?

https://docs.krustlet.dev/howto/krustlet-on-minikube/#step-3-install-and-run-krustlet

```
$ KUBECONFIG=~/.krustlet/config/kubeconfig \
  krustlet-wasi \
  --node-ip 10.0.2.2 \
  --node-name=krustlet \
  --bootstrap-file=${HOME}/.krustlet/config/bootstrap.conf
```

**Comment by abiosoft** [comment]
You can start colima with `--network-address` flag and specify the IP address as node ip. That should work.

1. Start with network address
    ```
    colima start --network address
    ```
2. Check the ip address
    ```
    $ colima ls
    PROFILE    STATUS     ARCH       CPUS    MEMORY    DISK     RUNTIME    ADDRESS
    default    Running    aarch64    2       2GiB      60GiB    docker     192.168.106.2
    ```
3. Use the IP address as node ip
    ```
    KUBECONFIG=~/.krustlet/config/kubeconfig \
    krustlet-wasi \
    --node-ip 192.168.106.2 \
    --node-name=krustlet \
    --bootstrap-file=${HOME}/.krustlet/config/bootstrap.conf
    ```

**Reply by sunfuze** [reply]
Thanks for reply. If i start with `--kubernetes` `--network_address`,  it seems k3s start master node with the address, and progress blocking, --very-verbose logs show `lima kubectl cluster-info` always retry and timeout.
